packages/adminrouter/extra/src/lib/util.lua

```diff
@@ -56,12 +56,116 @@ end
 
 
 function util.table_len(tbl)
+    -- Return the length of the table, without the limitations of #tbl
+    -- approach, where nil value in the table terminates counting/acts as the
+    -- end of the table.
     local count = 0
     for _ in pairs(tbl) do count = count + 1 end
     return count
 end
 
 
+function util.reverse(tbl)
+    -- Reverse in-place given table
+    local tl = util.table_len(tbl)
+
+    for i=1, math.floor(tl / 2) do
+        local tmp = tbl[i]
+        tbl[i] = tbl[tl - i + 1]
+        tbl[tl - i + 1] = tmp
+    end
+end
+
+
+function util.extract_service_path_component(service_path, fieldsLimit)
+    -- Extract path component in normalized form, at given level.
+    --
+    -- This function extracts from given service path the normalized service id
+    -- at a given level, defined by fieldsLimit.For example:
+    --
+    -- service_path group1/jenkins/ver/important/path/foobar.js
+    --
+    -- will yield following results, depending on fieldsLimit parameter:
+    -- nil:
+    --   normalised_name: foobar.js.path.important.ver.jenkins.group1
```

Comment:
no, please notice that it's the last component of the path - `foobar.js`. The "inversion" happens on `/`.